### PR TITLE
DT-1398 CVEs, App Insights v2, Kotlin v1.4.21

### PR DESCRIPTION
**Pull request by mikehalmamoj**
And a small refactor.

### CODE REVIEW

File: gradle/wrapper/gradle-wrapper.properties [hunk 1]
```diff
@@ -1,5 +1,5 @@
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-6.7-all.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-6.7.1-bin.zip
```
Comment (mikehalmamoj) [inline gradle/wrapper/gradle-wrapper.properties:3]: A strange one - as Gradle is built on Kotlin 1.3.72, if you include the full distribution it pulls in the old version of Kotlin.  This upset Intellij as it now had 2 different version of Kotlin on the classpath - it went very red and advised that strange things could happen.

File: src/main/kotlin/uk/gov/justice/digital/hmpps/gradle/configmanagers/BaseConfigManager.kt [hunk 2]
```diff
@@ -36,8 +36,7 @@ class BaseConfigManager(override val project: Project) : ConfigManager {
   }
 
   private fun addDependencies() {
-    project.dependencies.add("implementation", "com.fasterxml.jackson.module:jackson-module-kotlin:2.11.2")
```
Comment (mikehalmamoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/gradle/configmanagers/BaseConfigManager.kt:39]: I just moved this to the KotlinPluginManager as it felt more homely.

File: src/test/kotlin/uk/gov/justice/digital/hmpps/gradle/functional/ProjectRunner.kt [hunk 3]
```diff
@@ -26,14 +28,6 @@ data class ProjectDetails(
   override fun toString(): String = projectName
 }
 
-fun createAndRunJar(projectDetails: ProjectDetails): Process {
```
Comment (mikehalmamoj) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/gradle/functional/ProjectRunner.kt:29]: Nothing's actually deleted from this file, just some refactoring and moved public methods to the top where they can be found more easily.

File: release-notes/2.1.0.md [hunk 4]
```diff
@@ -0,0 +1,16 @@
+# 2.1.0
+
+Upgrade to Kotlin 1.4.21.
```
Comment (petergphillips) [inline release-notes/2.1.0.md:3]: might have been nice to link to kotlin release notes here

File: src/test/kotlin/uk/gov/justice/digital/hmpps/gradle/functional/pluginmanagers/KotlinPluginManagerTest.kt [hunk 5]
```diff
@@ -11,7 +11,7 @@ import uk.gov.justice.digital.hmpps.gradle.functional.buildProject
 import uk.gov.justice.digital.hmpps.gradle.functional.kotlinProjectDetails
 import uk.gov.justice.digital.hmpps.gradle.functional.makeProject
 
-class KotlinPluginManagerTest : GradleBuildTest(){
+class KotlinPluginManagerTest : GradleBuildTest() {
```
Comment (petergphillips) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/gradle/functional/pluginmanagers/KotlinPluginManagerTest.kt:14]: odd that this wasn't picked up by ktlint?

**Review by petergphillips — APPROVED** [review]
LGTM

File: src/test/kotlin/uk/gov/justice/digital/hmpps/gradle/functional/pluginmanagers/KotlinPluginManagerTest.kt (hunk 5 above)
Comment (mikehalmamoj) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/gradle/functional/pluginmanagers/KotlinPluginManagerTest.kt:14]: Interesting - will investigate.

**Comment by mikehalmamoj** [comment]
Spring Boot 2.4.1 is out and fixes the CVEs, so I'm going to upgrade to that instead.

**Pull request closed without merging**